tests/test_split.py

```diff
@@ -117,8 +99,13 @@ def test_find_breaks_error_if_no_h2_but_h1_headers(tmp_empty):
     'only H2 headings are supported at this time.' in str(excinfo.value)
 
 
-def test_find_breaks_error_if_no_h1_and_h2_headers(tmp_empty):
-    nb = _read(no_h1_and_h2_headers)
+@pytest.mark.parametrize('md', [
```

Comment:
Nice, looks way better.
You can also add this test into it: test_find_breaks_warning_if_only_one_h2_header
Just make sure you parameterize the string you're looking in the output as well.

Reply:
Do we want to merge `test_find_breaks_error_if_no_h2_but_h1_headers` `test_find_breaks_error_if_no_h1_and_h2_headers` `test_find_breaks_warning_if_only_one_h2_header` into one method, and put the warning texts as variables before methods?

Reply:
@94rain yes exactly! Any test case that repeats this pattern.

Reply:
I put them into two methods `test_find_breaks_errors` and `test_find_breaks_warnings` as they cannot be handled with the same logic.